pkg/fileutils/directory.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright The CloudNativePG Contributors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fileutils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"syscall"
+
+	"github.com/thoas/go-funk"
+
+	"github.com/cloudnative-pg/cloudnative-pg/pkg/management/log"
+)
+
+const (
+	createFileBlockSize = 262144
+	probeFileName       = "_cnpg_probe_"
+)
+
+type fileCreatorFunc = func(ctx context.Context, name string, size int) error
+
+// Directory represents a filesystem directory and provides methods to interact
+// with it, such as checking for available disk space by attempting to create
+// a file of a specified size.
+type Directory struct {
+	path           string
+	createFileFunc fileCreatorFunc
+}
```

Comment:
sorry, yet another comment.
I think this name, Directory, for this object, is far too wide.
Especially since this would be `fileutils.Directory` and there is a `fileutils.EnsureParentDirectoryExist` function.
This Directory is not a general directory implementation but a very specific use case.

Reply:
```suggestion
type DiskProbe struct {
	path           string
	createFileFunc fileCreatorFunc
}
```